### PR TITLE
MOB-2711 - Adyen drop-in integration working payments call

### PR DESCRIPTION
**Pull request by jsm20**
## JIRA
[MOB-2716](https://karhoo.atlassian.net/browse/MOB-2716)

## DESCRIPTION
Working payments call and fix for Adyen profile screen. Still more PRs to come!

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [x] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [x] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
N/a

### CODE REVIEW

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/payment/adyen/AdyenDropInService.kt [hunk 1]
```diff
@@ -51,9 +47,34 @@ class AdyenDropInService : DropInService() {
 
     // Handling for submitting additional payment details
     override fun makeDetailsCall(actionComponentData: JSONObject): CallResult {
+        val payload = JSONObject()
```
Comment (mohadian) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/payment/adyen/AdyenDropInService.kt:50]: Would be nice to move the logic outside this class to ensure SRP, could be a presenter/view and treat this class as the View. 
I understand we are trying to make it work and it is easier to keep it here for the time being

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/payment/adyen/AdyenDropInService.kt [hunk 2]
```diff
@@ -14,7 +14,7 @@ class AdyenDropInService : DropInService() {
     }
 
     private fun getPayment(paymentComponentData: JSONObject) {
-        val requestString = createPaymentRequestString(paymentComponentData.getJSONObject("paymentMethod"))
+        val requestString = createPaymentRequestString(paymentComponentData)
         KarhooApi.paymentsService.getAdyenPayments(requestString).execute { result ->
```
Comment (mohadian) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/payment/adyen/AdyenDropInService.kt:18]: Would be nice to move the logic outside this class to ensure SRP, could be a presenter/view and treat this class as the View.
I understand we are trying to make it work and it is easier to keep it here for the time being

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/payment/adyen/AdyenDropInService.kt (hunk 1 above)
Comment (jsm20) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/payment/adyen/AdyenDropInService.kt:50]: Yes. Agreed. Still not sure how everything fits together so keeping it here for now.

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/payment/adyen/AdyenDropInService.kt (hunk 2 above)
Comment (jsm20) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/payment/adyen/AdyenDropInService.kt:18]: As above

File: app/src/main/java/com/karhoo/karhootraveller/presentation/profile/ProfileActivity.kt [hunk 3]
```diff
@@ -81,6 +82,11 @@ class ProfileActivity : BaseActivity(), PaymentMVP.CardActions, UserProfileMVP.A
         }
     }
 
+    override fun onStart() {
+        super.onStart()
+        bookingPaymentDetailsWidget.bindPaymentDetails(KarhooApi.userStore.savedPaymentInfo, null)
```
[the remark this answers is not in the document]
Comment (jsm20) [reply app/src/main/java/com/karhoo/karhootraveller/presentation/profile/ProfileActivity.kt]: I've moved this into BookingPaymentView/Presenter to set up the default view on init.

**Review by mohadian — APPROVED** [review]

**Pull request merged**